python/bindings/py_decoded_message.hpp

```diff
@@ -32,19 +40,27 @@ struct PyDecodedMessage
 
     std::vector<FieldContainer> fields;
 
-    // MetaDataStruct with all fields that are no longer relevant after decoding dropped.
-    uint16_t message_id;
-    uint32_t message_crc;
-    std::string message_name;
-    PyGpsTime time;
-    MEASUREMENT_SOURCE measurement_source{MEASUREMENT_SOURCE::PRIMARY};
-    CONSTELLATION constellation{CONSTELLATION::UNKNOWN};
-
   private:
     mutable nb::dict cached_values_;
     mutable nb::dict cached_fields_;
 
     PyMessageDatabase::ConstPtr parent_db_;
 };
 
-} // namespace novatel::edie::oem
\ No newline at end of file
+
+//============================================================================
+//! \class PyMessage
+//! \brief A python representation for a single log message.
+//! 
+//! Extends PyField with reference to the Python represenation of a Header.
+//============================================================================
+struct PyMessage : public PyField
+{
+  public:
+    nb::object header;
```

Comment:
```suggestion
    oem::IntermediateHeader header;
```
I think you can avoid a tiny bit of indirection here and embed the header directly inside the PyMessage. Makes it a bit more cache-friendly.

Reply:
Good suggestion! I had to introduce a `PyHeader` subclass of `IntermediateHeader` to get the `Message.to_dict` method to still work, but I actually think this improves the code organization.